rclrs/src/client.rs

```diff
@@ -9,32 +9,33 @@ use rosidl_runtime_rs::Message;
 
 use crate::error::{RclReturnCode, ToResult};
 use crate::MessageCow;
-use crate::{rcl_bindings::*, RclrsError};
+use crate::{rcl_bindings::*, NodeHandle, RclrsError, ENTITY_LIFECYCLE_MUTEX};
 
 // SAFETY: The functions accessing this type, including drop(), shouldn't care about the thread
 // they are running in. Therefore, this type can be safely sent to another thread.
 unsafe impl Send for rcl_client_t {}
 
 /// Internal struct used by clients.
```

Comment:
I think any of these `_Handle` structs should have some documentation explaining what the intended (local) drop order is. 

i.e here its 
`Client` -> `Node`

But internally, the node handle of course expects context to drop first. So that could be omitted from here.

Reply:
> the node handle of course expects context to drop first

I'm not sure what you mean by this. The context needs to be the last thing that gets dropped, and that's what the current `[_]Handle` design is ensuring.

But I've added documentation for all the `[_]Handle` structs in [ae6d5b8](https://github.com/ros2-rust/ros2_rust/pull/386/commits/ae6d5b895828e65dac89c0960d45430707a6a3a9) so I'll mark this as resolved.

Reply:
The documentation you added is what I was looking for, thanks!